File: app/src/main/java/play/io/MediaPlayerActivity.java
package play.io;

import androidx.appcompat.app.AppCompatActivity;

import android.content.res.AssetFileDescriptor;
import android.media.MediaPlayer;
import android.os.Bundle;
import android.util.Log;

import java.io.IOException;

public class MediaPlayerActivity extends AppCompatActivity {

    private MediaPlayer mMediaPlayer;
    private final static String TAG = MainActivity.class.getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        getSupportActionBar().setTitle(MediaPlayerActivity.class.getSimpleName());
        mMediaPlayer = new MediaPlayer();
        playAssetsAudio("music.mp3");
    }

    private void playRawAudio() {
        mMediaPlayer = MediaPlayer.create(this, R.raw.music);
        mMediaPlayer.start(); // no need to call prepare(); create() does that for you
    }

    private void playAssetsAudio(final String name) {
        AssetFileDescriptor fd = null;
        try {
            fd = getAssets().openFd(name);
            mMediaPlayer = new MediaPlayer();
            mMediaPlayer.reset();
            mMediaPlayer.setDataSource(fd.getFileDescriptor(), fd.getStartOffset(), fd.getLength());
            mMediaPlayer.prepareAsync();
            mMediaPlayer.setOnPreparedListener((MediaPlayer mediaPlayer) -> {
                Log.d(TAG, "onPrepared: " + name);
                mediaPlayer.start();
                Log.d(TAG, "is playing: " + mediaPlayer.isPlaying());
            });
        } catch (Exception e) {
            Log.d(TAG, "catch error");
            try {
                if (fd != null) {
                    fd.close();
                }
            } catch (Exception e1) {
                Log.e(TAG, "Exception close fd: ", e1);
            }
        } finally {
            Log.d(TAG, "finally");
            if (fd != null) {
                try {
                    fd.close();
                } catch (IOException e) {
                    Log.e(TAG, "Finally, close fd ", e);
                }
            }
        }
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mMediaPlayer.stop();
        mMediaPlayer.release();
    }
}

File: app/src/main/java/android/playio/activities/MediaPlayerActivity.kt
package android.playio.activities

import android.content.res.AssetFileDescriptor
import android.media.MediaPlayer
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import android.playio.R
import java.io.IOException

class MediaPlayerActivity : AppCompatActivity() {
    private lateinit var mMediaPlayer: MediaPlayer
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_sample_navigation)
        supportActionBar!!.title = MediaPlayerActivity::class.java.simpleName
        mMediaPlayer = MediaPlayer()
        playAssetsAudio("audio/music.mp3")
    }

    private fun playRawAudio() {
//        mMediaPlayer = MediaPlayer.create(this, R.raw.music)
        mMediaPlayer.start() // no need to call prepare(); create() does that for you
    }

    private fun playAssetsAudio(name: String) {
        var fd: AssetFileDescriptor? = null
        try {
            fd = assets.openFd(name)
            mMediaPlayer = MediaPlayer()
            mMediaPlayer.reset()
            mMediaPlayer.setDataSource(fd.fileDescriptor, fd.startOffset, fd.length)
            mMediaPlayer.prepareAsync()
            mMediaPlayer.setOnPreparedListener { mediaPlayer ->
                Log.d(TAG, "onPrepared: $name")
                mediaPlayer.start()
                Log.d(TAG, "is playing: " + mediaPlayer.isPlaying)
            }
        } catch (e: Exception) {
            Log.d(TAG, "catch error: ${e.message}")
            try {
                fd?.close()
            } catch (e1: Exception) {
                Log.e(TAG, "Exception close fd: ", e1)
            }
        } finally {
            Log.d(TAG, "finally")
            if (fd != null) {
                try {
                    fd.close()
                } catch (e: IOException) {
                    Log.e(TAG, "Finally, close fd ", e)
                }
            }
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        mMediaPlayer.stop()
        mMediaPlayer.release()
    }

    companion object {
        private val TAG = SampleNavigationActivity::class.java.simpleName
    }
}